PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/survey/CustomerSurvey.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+package com.microsoft.azure.toolkit.intellij.common.survey;
+
+import com.intellij.ide.plugins.PluginManagerCore;
+import com.intellij.openapi.application.ApplicationInfo;
+import com.intellij.openapi.extensions.PluginDescriptor;
+import com.intellij.openapi.extensions.PluginId;
+import com.intellij.openapi.util.IconLoader;
+import com.microsoft.azure.toolkit.lib.common.telemetry.AzureTelemeter;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.apache.commons.lang.StringUtils;
+
+import javax.swing.*;
+import java.util.Optional;
+
+@Getter
+@AllArgsConstructor
+public enum CustomerSurvey implements ICustomerSurvey {
+
+    AZURE_SDK(Constants.AZURE_SDK_SURVEY_LINK, Constants.AZURE_SDK_SURVEY_DESCRIPTION, Constants.AZURE_ICON) {
+        @Override
+        public String getLink() {
+            return String.format(super.getLink(), getInstallationId());
+        }
+    },
+    AZURE_CLIENT_SDK(Constants.AZURE_CLIENT_SDK_SURVEY_LINK, Constants.AZURE_CLIENT_SDK_SURVEY_DESCRIPTION, Constants.AZURE_ICON) {
+        @Override
+        public String getLink() {
+            return String.format(super.getLink(), getInstallationId());
+        }
+    },
+    AZURE_MGMT_SDK(Constants.AZURE_MGMT_SDK_SURVEY_LINK, Constants.AZURE_MGMT_SDK_SURVEY_DESCRIPTION, Constants.AZURE_ICON) {
+        @Override
+        public String getLink() {
+            return String.format(super.getLink(), getInstallationId());
+        }
+    },
+    AZURE_SPRING_SDK(Constants.AZURE_SPRING_SDK_SURVEY_LINK, Constants.AZURE_SPRING_SDK_SURVEY_DESCRIPTION, Constants.AZURE_ICON),
+    AZURE_INTELLIJ_TOOLKIT(Constants.INTELLIJ_TOOLKIT_SURVEY_LINK, Constants.INTELLIJ_TOOLKIT_SURVEY_DESCRIPTION, Constants.AZURE_ICON) {
+        @Override
+        public String getLink() {
+            final String toolkit = Optional.ofNullable(PluginManagerCore.getPlugin(PluginId.getId("com.microsoft.tooling.msservices.intellij.azure")))
+                    .map(PluginDescriptor::getVersion).orElse(StringUtils.EMPTY);
+            final ApplicationInfo applicationInfo = ApplicationInfo.getInstance();
+            final String ide = String.format("%s %s", applicationInfo.getFullVersion(), applicationInfo.getBuild());
+            final String os = System.getProperty("os.name");
+            final String jdk = String.format("%s %s", System.getProperty("java.vendor"), System.getProperty("java.version"));
+            return String.format(super.getLink(), toolkit, ide, os, jdk, getInstallationId());
```

Comment:
Is it possible to let the output of super.getLink() as a constant ?

Reply:
super.getLink() will return the link value in constructor, which refers from constant.